auto_update.py

```diff
@@ -0,0 +1,56 @@
+import os
+import requests
+from ruamel.yaml import YAML
+
+def get_latest_grid_driver():
+    # URL of the JSON file containing driver information
+    url = "https://raw.githubusercontent.com/Azure/azhpc-extensions/refs/heads/master/NvidiaGPU/Nvidia-GPU-Linux-Resources.json"
+    response = requests.get(url)
+    response.raise_for_status()  
+    data = response.json()
+    
+    # Extract the latest GRID driver information
+    grid_versions = data['Latest']['Category']
+    grid_info = next((item for item in grid_versions if item["Name"] == "GRID"), None)
+    
+    if grid_info:
+        latest_version_info = grid_info['Versions'][0]
+        latest_version = latest_version_info['DriverVersion']
+        latest_url = latest_version_info['Driver'][0]['DirLink']
+        return latest_version, latest_url
+    
+    raise Exception("Could not find latest GRID driver version")
+
+# Add this at the end of your update_driver_config function
+def update_driver_config():
+    yaml = YAML()
+    yaml.preserve_quotes = True
+    yaml.indent(mapping=2, sequence=4, offset=2)
+
+    if not os.path.exists("driver_config.yml"):
+        raise FileNotFoundError("driver_config.yml not found in the current directory.")
+    
+    with open("driver_config.yml", "r") as f:
+        config = yaml.load(f)
+    
+    # Get latest version and URL
+    latest_version, latest_url = get_latest_grid_driver()
+    
+    # Store original version for comparison
+    original_version = config['grid']['version']
+    
+    # Update the grid section while preserving order
+    config['grid']['version'] = latest_version
+    config['grid']['url'] = latest_url
+    
+    # Write back to file
+    with open("driver_config.yml", "w") as f:
```

Comment:
you update in place the version/url and you also create a new_version.txt

Why do we need the new_version.txt if the orignal file is also updated ?

Reply:
To track the changed version in the Github action: for the PR title and branch name. It is a bit of a hacky way but it was a quick way to track changes. And then I delete the file in a subsequent step in the Github action. 

The other option would be to write more elaborate scripts in the Github action to track the version before/after the change.

Reply:
when doing the code Review, this is black magic to me. I would have normally changed this logic thinking there is a logical bug is this redundency isn't needed.   

This needs clarification

Reply:
fair, I'll try to simplify this or add comments to clarify.

Reply:
Noticed that I hadn't updated the changes from my fork to this branch properly earlier, sorry - that would've caused the confusion (as it didn't have the reference to the new file).

Reply:
Removed that new file logic from auto_update.py and now I find the version based on the diff in the Github action itself.